workflows/differentialabundance.nf

```diff
@@ -313,15 +312,25 @@ workflow DIFFERENTIALABUNDANCE {
             VALIDATOR.out.feature_meta.map{ it[1] },
             [params.features_id_col, params.features_name_col]    
         )
-    
+
+        // Make channel from current file and get it's name for correct saving of output
+
+        ch_gene_sets = Channel.fromList(gene_sets_files)
+
         // The normalised matrix does not always have a contrast meta, so we
         // need a combine rather than a join here
+        // Also add file name to metamap for easy access from modules.config
 
         ch_gsea_inputs = CUSTOM_TABULARTOGSEAGCT.out.gct
             .map{ it.tail() }
             .combine(CUSTOM_TABULARTOGSEACLS.out.cls)
             .map{ tuple(it[1], it[0], it[2]) }
             .combine(ch_gene_sets)
+            .map{ tuple( ["gmt_name": it[3].getName().split("\\.")[0..-2].join("."),
+                            "reference": it[0].reference,
+                            "target": it[0].target,
+                            "id": it[0].id],
+                        it[1], it[2], it[3]) }
```

Comment:
```suggestion
```
Apologies, but I think we can lose this entirely, it's only used for the modules.config, which I _think_ can reference its gene_sets input directly